src/project-type/workspace-instance.ts

```diff
@@ -0,0 +1,34 @@
+import { WorkspaceApi, WorkspaceImpl } from "@sap/project-api";
+import { BasWorkspaceApi } from "../../types/api";
+
+let basWSAPI: BasWorkspaceApi;
+
+export function getBasWorkspaceAPI(): BasWorkspaceApi {
+  if (!basWSAPI) {
+    const workspaceImpl = new WorkspaceImpl();
+
+    initWorkspaceImpl(workspaceImpl);
+
+    basWSAPI = createWorkspaceReadOnlyProxy(workspaceImpl);
+  }
+
+  return basWSAPI;
+}
+
+
+function initWorkspaceImpl(workspaceImpl: WorkspaceApi): void {
+  workspaceImpl.startWatch();
```

Comment:
Should `startWatch()`  happen after `getBasWorkspaceAPI()` is called?
But we need the "watching" to work also for internal logic (e.g the VSCode contexts watching logic).
So maybe it should be de-coupled from the **External** BasWsApi" logic